salt/metalk8s/defaults.yaml

```diff
@@ -113,3 +113,12 @@ coredns:
 upgrade: false        # define if we're on an upgrade case
 
 proxies: {}
+
+certificates:
+  beacon:
+    notify_days: 45
+    interval: 86400  # once a day
+  days_valid: 365
+  days_remaining: 90
```

Comment:
Should these really be set globally?

Reply:
You want to set these per certificate ? I also asked myself on this one, but I'm not sure it will be useful.
Otherwise we could do both and fall back on the global value if nothing is specified at a specific certificate level.
e.g. in `pillar/metalk8s/roles/bootstrap.sls`:
```
certificates:
  kubeconfig:
    salt-master:
      path: /etc/salt/master-kubeconfig.conf
      cn: salt-master-{{ grains.id }}
      days_remaining: 60
      days_valid: 180
      regen_sls:
        - metalk8s.salt.master.kubeconfig
[...]
```
Then in the formula `salt/metalk8s/salt/master/kubeconfig.sls` we could simply do:
```
[...]
    - days_valid: {{ certificates.kubeconfig['salt-master'].days_valid | default(certificates.days_valid) }}
    - days_remaining: {{ certificates.kubeconfig['salt-master'].days_remaining | default(certificates.days_remaining) }}
[...]
```

Reply:
Maybe just rename the root key (or subkey) so that here we only talk about "client certificates" and not CAs.
And for me, we may want at some point to change the one in some kubeconfig so it could be useful to override in the pîllar (or at least having different key for "client certificates", "kubeconfig certificates", "CA certificates")

Reply:
Yeah, maybe just build a macro to handle the defaulting, and indeed, maybe add a level to distinguish between cert "kinds" (CA - server - client - kubeconfig). Granted though, this will likely not be very useful to have this setting available per-cert, except for some of them (e.g. an admin could want to extend the duration of their `/etc/k/admin.conf` to avoid having to regenerate it every year).

Reply:
I've already changed this, we now have:
```
certificates:
  beacon:
    notify_days: 45
    interval: 86400
  client:
    days_remaining: 90
    days_valid: 365
    files:
      /path/to/client.crt:
        days_valid: 180
        [...]
  kubeconfig:
    days_remaining: 90
    days_valid: 365
    files:
      /path/to/kubeconfig.conf:
        [...]
```
It uses the configuration set under `certficates.[kubeconfig|client].files.$path` if any and fallback on `certificates.[kubeconfig|client]` otherwise.
I've put everything that's not CA nor kubeconfig under client, but maybe we want to distinguish client & server certificates ?

Reply:
Yes maybe add one section for server, right, it make sense